File: src/main/java/com/cds/command/CommandHandler.java
package com.cds.command;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.ConsoleCommandSender;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;

import com.cds.handler.BaseHandler;

public class CommandHandler extends BaseHandler {
    private ConsoleCommandHandler consoleCommands = new ConsoleCommandHandler(plugin);
    private GameCommandHandler gameCommandHandler = new GameCommandHandler(plugin);

    public CommandHandler(JavaPlugin plugin) {
        super(plugin);
    }

    public boolean handleCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (sender instanceof ConsoleCommandSender)
            return consoleCommands.handleCommand((ConsoleCommandSender) sender, cmd, label, args);
        if (sender instanceof Player)
            return gameCommandHandler.handleCommand((Player) sender, cmd, label, args);
        return false;
    }
}

File: src/main/kotlin/com/cds/command/CommandHandler.kt
package com.cds.command

import com.cds.handler.BaseHandler
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.ConsoleCommandSender
import org.bukkit.entity.Player
import org.bukkit.plugin.java.JavaPlugin

public class CommandHandler(plugin: JavaPlugin) : BaseHandler(plugin) {
    val consoleCommands = ConsoleCommandHandler(plugin)
    val gameCommandHandler = GameCommandHandler(plugin)

    fun handleCommand(
            sender: CommandSender,
            cmd: Command,
            label: String,
            args: Array<String>
    ): Boolean {
        if (sender is ConsoleCommandSender)
                return consoleCommands.handleCommand(sender, cmd, label, args)
        if (sender is Player) return gameCommandHandler.handleCommand(sender, cmd, label, args)
        return false
    }
}
